docker-compose.yml

```
@@ -1,13 +1,82 @@
+# This docker-compose.yml sets up a deployment of Mastodon's web (Puma), Sidekiq, and streaming containers,
+# as well as the required Postgres and Redis components. Optionally, you can enable Elastic Search and Tor by
+# removing the comments (#) from those blocks.
+
+# You should check the Docker image tag used for the web, sidekiq, and streaming containers to make sure you are running
+# the appropriate version for your deployment. Using `nightly` will deploy the in-development version of Mastodon, which is
+# not considered a stable release. Visit https://github.com/mastodon/mastodon/releases for currently released versions and tags.
+
 version: '3'
 services:
-  db:
+  web:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:nightly
+    env_file: .env.production
     restart: always
-    image: postgres:14-alpine
-    shm_size: 256mb
+    command: bundle exec puma -C config/puma.rb
+    healthcheck:
+      test: ['CMD-SHELL', "curl -s --noproxy localhost localhost:3000/health | grep -q 'OK' || exit 1"]
+    networks:
+      - external_network
+      - internal_network
+    ports:
+      - '127.0.0.1:3000:3000'
+    volumes:
+      - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      # - es
+
+  sidekiq:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:nightly
+    env_file: .env.production
+    restart: always
+    command: bundle exec sidekiq
+    healthcheck:
+      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+    networks:
+      - external_network
+      - internal_network
+    volumes:
+      - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      # - es
+
+  streaming:
+    # build:
+    #   context: ./streaming
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon-streaming:nightly
+    env_file: .env.production
+    restart: always
+    command: node ./streaming
```

```suggestion
    command: node ./streaming/index.js
```
Whilst node will understand just the folder, it's better to give an explicit path, I fully expect this to change when we restructure the streaming codes filesystem, e.g., to `streaming/bin/server.js`

---

Oops, I had thought I included this as we've discussed it before.